File: Completed/python3/1331.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
      setRank = {}
      for v in arr:
        setRank[v] = 0
        
      ranks = []
      for k in setRank:
        ranks.append(k)
      
      ranks.sort()
      for i, val in enumerate(ranks):
        setRank[val] = i+1
      
      for i, val in enumerate(arr):
        arr[i] = setRank[val]
      return arr
      
    
        

File: Completed/Go/1331.go
import (
  "sort"
)

func arrayRankTransform(arr []int) []int {
  setRank := map[int]int{}
  for _, v := range arr {
    setRank[v] = 0
  }
  ranks := make([]int, 0, len(setRank))
  for k := range setRank {
    ranks = append(ranks, k)
  }
  sort.Ints(ranks)
  for i, k := range ranks {
    setRank[k] = i+1
  }
  
  for i, v := range arr {
    arr[i] = setRank[v]
  }
  
  return arr
}
